### PR TITLE
Refactor exercises 5 and 6 of Echidna

### PR DESCRIPTION
**Pull request by montyly**
- Invert 5 with 6 (the CTF order does not follow the Echidna's difficulty order)
- Create a separate repo with everything configured: https://github.com/crytic/damn-vulnerable-defi-echidna
  - There are three branches, master, hints, solutions, to allow everyone following their own progression
- Add a hints section, with templates and guidelines (ex. the multi-abi was never introduced)
- Remove the complex setup due to the changes in damn vulnerable defi (everything is done in the separate repo)

There is no CI tests yet, but it should be simple to do it using the `solutions` branch

### CODE REVIEW

**Comment by montyly** [comment]
cc @anishnaik / @ggrieco-tob 

Relevant PR: https://github.com/crytic/building-secure-contracts/pull/143

**Comment by ggrieco-tob** [comment]
This looks good in general (I still need to review in detail). I think @elopez can add some basic CI tests before merging.

File: program-analysis/echidna/Exercise-6.md [hunk 1]
```diff
@@ -4,63 +4,79 @@
...
+
+Now, the two balance trackers are out-of-sync.
+
+When Echidna calls `pool.flashLoan(10)`, the assertion `assert(poolBalance == balanceBefore)` in `UnstoppableLender` will break and the pool can no longer provide flash loans.
 
 See example output below from Echidna:
 
 ```bash
-$ npx hardhat clean && npx hardhat compile --force && echidna-test . --contract TestNaiveReceiverEchidna --multi-abi --config contracts/naive-receiver/config.yaml
+$ echidna-test . --contract UnstoppableEchidna --config contracts/unstoppable/config.yaml
```
Comment (elopez) [inline program-analysis/echidna/Exercise-6.md:72]:
```suggestion
$ echidna-test . --contract UnstoppableEchidna --config unstoppable.yaml
```

File: program-analysis/echidna/Exercise-6.md [hunk 2]
```diff
@@ -4,63 +4,79 @@
...
+- The `receiveTokens` callback function must be implemented
+- A template is provided in [contracts/unstoppable/UnstoppableEchidna.sol](https://github.com/crytic/damn-vulnerable-defi-echidna/blob/hints/contracts/unstoppable/UnstoppableEchidna.sol)
+- A config file is provided in [unstoppable.yaml](https://github.com/crytic/damn-vulnerable-defi-echidna/blob/hints/unstoppable.yaml)
+
+
+
 ## Solution
 
-This solution can be found in [exercises/exercise6/solution.sol](./exercises/exercise6/solution.sol)
+This solution can be found in [`solutions` branch](https://github.com/crytic/damn-vulnerable-defi-echidna/blob/hints/contracts/unstoppable/UnstoppableEchidna.sol).
```
Comment (elopez) [inline program-analysis/echidna/Exercise-6.md:49]:
```suggestion
This solution can be found in the [`solutions` branch](https://github.com/crytic/damn-vulnerable-defi-echidna/blob/hints/contracts/unstoppable/UnstoppableEchidna.sol).
```

**Review by elopez — COMMENTED** [review]
CI tests added 👍 I also left a few fixes and improvements as comments below

**Pull request merged**